flink/v1.18/flink/src/main/java/org/apache/iceberg/flink/source/IcebergTableSource.java

```diff
@@ -131,16 +131,20 @@ private DataStream<RowData> createDataStream(StreamExecutionEnvironment execEnv)
   private DataStreamSource<RowData> createFLIP27Stream(StreamExecutionEnvironment env) {
     SplitAssignerType assignerType =
         readableConfig.get(FlinkConfigOptions.TABLE_EXEC_SPLIT_ASSIGNER_TYPE);
-    IcebergSource<RowData> source =
+    IcebergSource.Builder builder =
         IcebergSource.forRowData()
             .tableLoader(loader)
-            .assignerFactory(assignerType.factory())
             .properties(properties)
             .project(getProjectedSchema())
             .limit(limit)
             .filters(filters)
-            .flinkConfig(readableConfig)
-            .build();
+            .flinkConfig(readableConfig);
+
+    if (assignerType != null) {
```

Comment:
why is this changed? assigner type has a default type. old code should be fine, right?